src/pybind/mgr/cephadm/serve.py

```diff
@@ -390,10 +390,9 @@ def _check_for_strays(self) -> None:
                 for s in daemons:
                     name = '%s.%s' % (s.get('type'), s.get('id'))
                     if s.get('type') == 'rbd-mirror':
-                        defaults = defaultdict(lambda: None, {'id': None})
                         metadata = self.mgr.get_metadata(
-                            "rbd-mirror", s.get('id'), default=defaults)
-                        if metadata['id']:
```

Comment:
lgtm. Note that this code block is not very well tested. Neither in pytest nor in teuthology.

Reply:
ack.